### PR TITLE
upgraded ci to test and build on multi platforms

### PR DESCRIPTION
**Pull request by aviadtamir**

### CODE REVIEW

File: .github/workflows/test-build.yml [hunk 1]
```diff
@@ -7,26 +7,94 @@ on:
...
+      matrix:
+        dotnet-version: [3.1.301]
+        os:  [macOS-latest]
+        tests-paths: [./BruteShark/PcapAnalyzerTest, ./BruteShark/PcapProcessorTest]
+        projects-paths: [./BruteShark/BruteSharkCli]
+
+    steps:
+    - uses: actions/checkout@v2
+    
+    - name: Setup .NET Core
```
Comment (odedshimon) [inline .github/workflows/test-build.yml:24]: The code of "Setup", "Install Dependencies", "Test" and "Build" are the same, it worth to search a way to use functions for less code duplication and more readability

File: BruteShark/PcapAnalyzer/Modules/PasswordsModule/CredentialsParsers/FtpPasswordParser.cs [hunk 2]
```diff
@@ -6,8 +6,8 @@
 namespace PcapAnalyzer
 {
     public class FtpPasswordParser : IPasswordParser
-    {
-        private Regex _ftpSuccessfullLoginRegex = new Regex(@"220(.*)\r\nUSER\s(?<Username>.*)\r\n331(.*)\r\nPASS\s(?<Password>.*)\r\n");
+    {   
+        private Regex ftpSuccessfullLoginRegex = new Regex(@"220(.*)"+Environment.NewLine+ @"USER\s(?<Username>.*)"+Environment.NewLine+@"331(.*)" + Environment.NewLine + @"PASS\s(?<Password>.*)" + Environment.NewLine);
```
Comment (odedshimon) [inline BruteShark/PcapAnalyzer/Modules/PasswordsModule/CredentialsParsers/FtpPasswordParser.cs:10]: There is a problem here: we use the user operating system new line format, but the data we get can be from any OS. I guess that was a potential bug before this PR.  I believe a Regex that catch both formats will be better

**Review by odedshimon — COMMENTED** [review]
Very Nice :-)
DevOps Rocks!

File: BruteShark/PcapAnalyzer/Modules/PasswordsModule/CredentialsParsers/FtpPasswordParser.cs (hunk 2 above)
Comment (aviadtamir) [reply BruteShark/PcapAnalyzer/Modules/PasswordsModule/CredentialsParsers/FtpPasswordParser.cs:10]: you are right

File: .github/workflows/test-build.yml (hunk 1 above)
Comment (aviadtamir) [reply .github/workflows/test-build.yml:24]: they are indeed similar, bit there are differences between them, the only two steps that are common to all jobs are the checkout and the setup dotnet,
there is an option to put these two steps in another yaml and create a "composite" actions flow
https://github.blog/changelog/2020-08-07-github-actions-composite-run-steps/

**Pull request merged**